evm_loader/program/src/instruction/transaction_step.rs

```diff
@@ -134,6 +140,36 @@ fn allocate_or_reinit_state(
     Ok(())
 }
 
+fn handle_priority_fee<'a>(
+    storage: &StateAccount<'a>,
```

Comment:
In the first iteration you should burn `gas_limit*max_fee_per_gas` tokens from the origin account. 
These tokens will be locked in storage and you transfer gas from storage to operator in each iteration.
Return remaining tokens to the origin in the last iteration or on cancel.

See `StateAccount::consume_gas`

Reply:
yes, I did it as described in another comment.